mount_darwin.go

```diff
@@ -198,6 +198,28 @@ func callMount(bin string, daemonVar string, dir string, conf *mountConfig,
 		return nil, fmt.Errorf("mount_osxfusefs: %v", err)
 	}
 
+	// After calling the mount helper, we need to assume the volume has been
+	// mounted. So we need to call umount in case error happens when trying to
+	// receive the fd.
+	//
+	// This is important because otherwise since some file system operations
+	// are delayed until the FUSE volume has been initialized (FUSE_INIT), some
+	// system processes will be put on hold until the mount is complete or the
+	// volume is unmounted due to daemon_timeout. When it happens, Safari
+	// refuses to laod any page and reboot gets stuck as well.
+	//
+	// We don't have to call unmount if cmd.Start() returns an error because it
+	// only returns an error if starting the process fails.
+	defer func() {
+		if err != nil {
+			log.Printf(
+				"calling syscall.Unmount because callMount failed: %v", err)
+			if err := syscall.Unmount(dir, 0); err != nil {
+				log.Printf("syscall.Unmount failed: %v", err)
```

Comment:
Need a `\n` at the end?

Reply:
Nope! `log` functions appends one automatically if there isn't a trailing `\n` already: https://play.golang.org/p/GJ6ogvz-kRT

Reply:
D'oh thanks, misread it as `fmt.`.